rust/src/nasl/utils/hosts.rs

```diff
@@ -0,0 +1,20 @@
+// SPDX-FileCopyrightText: 2024 Greenbone AG
+//
+// SPDX-License-Identifier: GPL-2.0-or-later WITH x11vnc-openssl-exception
+
+use std::net::{SocketAddr, ToSocketAddrs};
+
+use crate::nasl::utils::error::FunctionErrorKind;
+
+pub fn resolve(
+    mut hostname: String,
+) -> Result<Option<Box<dyn Iterator<Item = SocketAddr>>>, FunctionErrorKind> {
+    //std::net to_socket_addrs() requires a port. Therefore, using a dummy port
+    hostname.push_str(":5000");
```

Comment:
Given that we add a dummy port, we shouldn't be returning `SocketAddr` from this function, but rather return an `IpAddr`, since that is the only valuable information a caller can extract.